session_rpcserver.go

```diff
@@ -20,33 +21,39 @@ type sessionRpcServer struct {
 
 	db            *session.DB
 	sessionServer *session.Server
+
+	superMacBaker func(ctx context.Context, rootKeyID uint64,
+		recipe *session.MacaroonRecipe) (string, error)
 }
 
 // AddSession adds and starts a new Terminal Connect session.
 func (s *sessionRpcServer) AddSession(_ context.Context,
 	req *litrpc.AddSessionRequest) (*litrpc.AddSessionResponse, error) {
 
-	var (
-		typ    session.Type
-		expiry time.Time
-	)
-	switch req.SessionType {
-	case litrpc.SessionType_TYPE_UI_PASSWORD:
-		typ = session.TypeUIPassword
+	typ, err := unmarshalRPCType(req.SessionType)
+	if err != nil {
+		return nil, err
+	}
 
-	default:
+	if typ != session.TypeUIPassword && typ != session.TypeMacaroonAdmin {
 		return nil, fmt.Errorf("invalid session type, only UI " +
-			"password supported in LiT")
+			"password, macaroon admin and macaroon readonly " +
+			"types supported in LiT")
 	}
 
-	expiry = time.Unix(int64(req.ExpiryTimestampSeconds), 0)
+	expiry := time.Unix(int64(req.ExpiryTimestampSeconds), 0)
 	if time.Now().After(expiry) {
 		return nil, fmt.Errorf("expiry must be in the future")
 	}
 
+	var macPerms []bakery.Op
+	if typ == session.TypeMacaroonAdmin {
```

Comment:
Unify this `if` over the `typ` with the one above and transform into a nice `switch` statement that combines both tasks?